crates/proof-of-sql/utils/generate-parameters/main.rs

```diff
@@ -0,0 +1,299 @@
+//! A CLI utility to generate the public parameters for the prover and verifier
+
+#[cfg(test)]
+mod round_trip_test;
+
+use ark_std::rand::SeedableRng;
+use clap::Parser;
+use indicatif::{ProgressBar, ProgressStyle};
+use proof_of_sql::proof_primitive::dory::{ProverSetup, PublicParameters, VerifierSetup};
+use rand_chacha::ChaCha20Rng;
+use sha2::{Digest, Sha256};
+use std::{
+    env,
+    fs::{self, File, OpenOptions},
+    io,
+    io::Write,
+    path::Path,
+    process::Command,
+    time::{Duration, Instant},
+};
+
+/// Transparent public randomness
+const SEED: &str = "SpaceAndTime";
+
+const BLITZAR_PARTITION_WINDOW_WIDTH: &str = "BLITZAR_PARTITION_WINDOW_WIDTH";
+
+// Command-line argument parser structure
+#[derive(Parser, Debug)]
+#[command(author, version, about, long_about = None)]
+struct Args {
+    /// The value for `nu` (number of public parameters)
+    #[arg(short, long, default_value_t = 8)]
+    nu: usize,
+
+    /// Mode for generating parameters: "p" for Prover, "v" for Verifier
+    #[arg(short, long, default_value = "pv")]
+    mode: String,
+
+    /// The initial randomness for the transparent setup
+    #[arg(short, long, default_value = SEED)]
+    seed: String,
+
+    /// The directory to store generated files and archives
+    #[arg(short, long, default_value = "./output")]
+    target: String,
+}
+
+fn main() {
+    // Set the BLITZAR_PARTITION_WINDOW_WIDTH environment variable
+    env::set_var(BLITZAR_PARTITION_WINDOW_WIDTH, "14");
+
+    // Confirm that it was set by reading it back
+    match env::var(BLITZAR_PARTITION_WINDOW_WIDTH) {
+        Ok(value) => {
+            println!("Environment variable {BLITZAR_PARTITION_WINDOW_WIDTH} set to {value}");
+        }
+        Err(e) => println!("Failed to set {BLITZAR_PARTITION_WINDOW_WIDTH}: {e}"),
```

Comment:
Are we supposed to exit if we can't set the env variable? If so let's use `std::process::exit(-1)`.